segmentstore/server/src/test/java/io/pravega/segmentstore/server/tables/ContainerKeyIndexTests.java

```diff
@@ -996,6 +996,45 @@ public void testRegularSegmentThrottling() {
         Assert.assertEquals(segmentTracker.getUnindexedSizeBytes(1L), TableExtensionConfig.MAX_UNINDEXED_LENGTH.getDefaultValue() - 1);
     }
 
+    @Test
+    public void testRecoveryWithBatchesAndPartialEntries() throws Exception {
```

Comment:
This process interacts in non-trivial ways with throttling. Ideally we would move throttling out into its own class. After all each class should have one purpose and currently the class is doing like 6 things. If it absolutely cannot be done, then we should add test coverage for the full powerset of operations/states.

Reply:
I agree that the class is too complex and does multiple things, but this PR is thought to be cherry picked up to r0.10 at least. For this reason I would like to keep changes at minimum, so we minimize risk of introducing new problems. I agree with the observation of adding more tests, will try to do that.